Firmware/firmwareLiBCM/src/LTC68042cell.cpp

```diff
@@ -239,48 +241,56 @@ void processAllCellVoltages(void)
 //latest validated results are stored in a different, globally-accessible array inside "LTC68042_result.c"
 //Example with QTY48 cells:
 //  -the absolute first call starts a conversion.
-//  After that, the behavior is as follows:  
+//  After that, the behavior is as follows:
 //  -the next sixteen calls ( (48 cells) / (3 cells per call) = 16 calls ) read back QTY48 cell voltages.
 //  -The seventeenth call performs all pack voltage math and stores valid results in LTC68042_result.c
 //
 //returns false while gathering data, true each time all data is processed
 bool LTC68042cell_nextVoltages(void)
 {
     static uint8_t presentState = LTC_STATE_FIRSTRUN;
+    static bool conversionInProcess = false; //used to speed up execution of conversion complete test
     bool cellVoltageDataStatus = GATHERING_CELL_DATA;
 
     if (LTC68042configure_wakeup() == LTC6804_CORE_JUST_WOKE_UP) { presentState = LTC_STATE_FIRSTRUN; }
 
     if (presentState == LTC_STATE_GATHER)
     { //retrieve next CVR from LTC, then validate and store in cellVoltages_counts[][] array
-
-        //round-robin state handlers
-        static uint8_t chipAddress = FIRST_IC_ADDR;
-        static char cellVoltageRegister = 'A'; //LTC68042 contains QTY4 CVRs (A/B/C/D)
-
-        validateAndStoreNextCVR(chipAddress, cellVoltageRegister);
-
-        //determine which LTC68042 IC & CVR to read next
-        cellVoltageRegister++;
-        if (cellVoltageRegister >= 'E')
-        { 
-            //LTC6804 only has registers A,B,C,D
-            cellVoltageRegister = 'A'; //reset back to first CVR
-
-            if (++chipAddress >= (FIRST_IC_ADDR + TOTAL_IC))
-            { 
-                //just finished reading last IC's last CVR... all cell voltages stored in cellVoltages_counts[][]
-                startCellConversion(); //start the next cell conversion //takes a while to finish
-                
-                chipAddress = FIRST_IC_ADDR; //reset to first LTC IC
-                presentState = LTC_STATE_PROCESS; //all cell voltages gathered.  Process data on next run.
+        // but don't gather data or advance the state if the  current conversion is not complete (should not usually be necessary)
+        if ( ( ! conversionInProcess) || (LTC6804_MAX_CONVERSION_TIME_ms < (millis() - conversionStart_ms)) )
```

Comment:
Per my other comment, this code will likely go away... but in general I prefer the following formatting:
-explicit boolean comparisons
-newline for each logical compare

```suggestion
        if ( (conversionInProcess == false)                                    ||
             (LTC6804_MAX_CONVERSION_TIME_ms < (millis() - conversionStart_ms)) )
```

Reply:
Fixed in next iteration